include/swift/AST/DiagnosticsSema.def

```diff
@@ -2710,6 +2710,9 @@ NOTE(missing_member_type_conformance_prevents_synthesis, none,
      "protocol %2, preventing synthesized conformance "
      "of %3 to %2",
      (unsigned, Type, Type, Type))
+NOTE(classes_equatable_hashable_synthesis,none,
+    "Currently 'Equatable' and 'Hashable' synthesis is not supported for classes "
```

Comment:
Please start the diagnostic text with a lowercase character.

Reply:
Also, I think you should only mention either Hashable or Equatable in the diagnostic text (depending on what’s being used), instead of mentioning both.

Reply:
Thanks. Fixed.

Reply:
Here’s what you need:

“automatic synthesis of '%0` is not supported for classes”. Then, in diagnoseFailedDerivation, you check which protocol is being used and then emit this diagnostic, passing the name of the protocol.

Reply:
Updated diagnostic. Fixed.